mesonbuild/build.py

```diff
@@ -790,6 +801,22 @@ def link(self, target):
                 raise InvalidArguments('Tried to mix cross built and native libraries in target {!r}'.format(self.name))
             self.link_targets.append(t)
 
+    def link_whole(self, target):
+        if not isinstance(target, list):
+            target = [target]
+        for t in target:
```

Comment:
Could we refactor this and the loop in link() into a helper that just took a bool of whether shared libs are allowed?